spec/jobs/batch_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'spec_helper'
+require 'spec/support/persistence'
+require 'spec/support/events'
+require 'spec/support/dummy'
+require 'spec/support/env'
+require 'routemaster/jobs/batch'
+require 'routemaster/services/codec'
+# require 'routemaster/services/deliver'
+
+module Routemaster
+  describe Jobs::Batch do
+    subject { described_class.new(delivery: delivery) }
+
+    let(:subscriber) {
+      Routemaster::Models::Subscriber.new(name: 'alice').tap do |s|
+        s.timeout = 0
+      end
+    }
+    let(:messages) { [make_event, make_event] }
+    let(:delivery) { double 'delivery', call:nil }
```

Comment:
I'm pretty sure if you did `instance_double(Services::Deliver, ...` then you'll get the class methods of that class. `instance_double` is badly named and doesn't really have anything to do with instances...